libgadget/blackhole.c

```diff
@@ -599,40 +945,73 @@ blackhole_accretion_ngbiter(TreeWalkQueryBHAccretion * I,
             }
         }
     }
+    
+    
+    /********************************************/
+    /* Find min pot pos within halo */
+    if(r < 10)
+    {
+        if(P[other].Potential < O->BH_HaloMinPot)
+        {
+            int d;
+            O->BH_HaloMinPot = P[other].Potential;
+            for(d = 0; d < 3; d++) {
+                O->BH_HaloMinPotPos[d] = P[other].Pos[d];
+            }
+        }
+    }
+    /********************************************/
 
     /* Accretion / merger doesn't do self interaction */
     if(P[other].ID == I->ID) return;
-
-    if(P[other].Type == 5 && r2 < iter->accretion_kernel.HH)	/* we have a black hole merger */
+    
+    /* we have a black hole merger, use 2 times GravitationalSoftening as merging criteria*/
+    if(P[other].Type == 5 && r < (2*FORCE_SOFTENING(0,1)/2.8))	
     {
-        /* We do not depend on the BH relative velocity.
-         * Because the BHs are not dissipative, their relative velocities
-         * can be large, causing clumps of BHs to build up
-         * at the same position without merging. */
-        MyIDType readid, newswallowid;
-
-        #pragma omp atomic read
-        readid = (BHP(other).SwallowID);
-
-        /* Here we mark the black hole as "ready to be swallowed" using the SwallowID.
-         * The actual swallowing is done in the feedback treewalk by setting Swallowed = 1
-         * and merging the masses.*/
-        do {
-            /* Generate the new ID from the old*/
-            if(readid != (MyIDType) -1 && readid < I->ID ) {
-                /* Already marked, prefer to be swallowed by a bigger ID */
-                newswallowid = I->ID;
-            } else if(readid == (MyIDType) -1 && P[other].ID < I->ID) {
-                /* Unmarked, the BH with bigger ID swallows */
-                newswallowid = I->ID;
-            }
-            else
-                break;
-        /* Swap in the new id only if the old one hasn't changed:
-         * in principle an extension, but supported on at least clang >= 9, gcc >= 5 and icc >= 18.*/
-        } while(!__atomic_compare_exchange_n(&(BHP(other).SwallowID), &readid, newswallowid, 0, __ATOMIC_RELAXED, __ATOMIC_RELAXED));
+        O->encounter = 1; // mark the event when two BHs encounter each other
```

Comment:
I wasn't so clear what this was for? It marks events where they are close, and is useful because sometimes they are close but do not merge? If so would it be useful to store it as a counter in the BHP struct?

Reply:
Yes, it gives me a quick estimate of how long does the Gravitational Bound criteria delay the merging event: (time difference between the first encounter>0 and acBHMass>0.)  It is already in BHP struct. By 'counter' you mean increment every time step?

Reply:
Yes, I mean increment every time it happens, so we can track objects that have multiple close encounters on the way to a merger.

Reply:
I am not sure how to do this. I think we need to distinguish between different P[other].ID to account the multiple encounters? Otherwise we are just increment 1 every time step.

Reply:
Hmm, yes, that is a good point. That would be quite difficult to do. Just keep it as it is then